File: electron/models/Inscription.js
import { v4 as uuidv4 } from 'uuid';

export class Inscription {
  constructor(data = {}) {
    this.id = data.id || uuidv4();
    this.anneeScolaire = data.anneeScolaire || data.annee_scolaire || '';
    this.dateInscription = data.dateInscription || data.date_inscription || new Date().toISOString().split('T')[0];
    this.statutScolaire = data.statutScolaire || data.statut_scolaire || 'nouveau';
    this.classeId = data.classeId || data.classe_id || '';
    this.eleveDataId = data.eleveDataId || data.eleve_data_id || '';
    this.createdAt = data.createdAt || data.created_at || new Date().toISOString();
    this.updatedAt = data.updatedAt || data.updated_at || new Date().toISOString();
    
    this.niveauClasse = data.niveauClasse || data.niveau_classe || '';
    this.classe = data.classe || '';
    this.cycle = data.cycle || '';
    this.niveauScolaire = data.niveauScolaire || data.niveau_scolaire || '';
    this.nom = data.nom || '';
    this.prenom = data.prenom || '';
    this.dateNaissance = data.dateNaissance || data.date_naissance || '';
    this.sexe = data.sexe || '';
    this.photo = data.photo || null;
    this.matricule = data.matricule || '';
    this.statut = data.statut || '';
    this.lieuDeNaissance = data.lieuDeNaissance || data.lieu_de_naissance || null;
    this.contact = data.contact || null;
    this.payements = data.payements || [];
  }

  toDB() {
    return {
      id: this.id,
      annee_scolaire: this.anneeScolaire,
      date_inscription: this.dateInscription,
      statut_scolaire: this.statutScolaire,
      classe_id: this.classeId,
      eleve_data_id: this.eleveDataId,
      created_at: this.createdAt,
      updated_at: this.updatedAt
    };
  }

  toJSON() {
    return {
      id: this.id,
      anneeScolaire: this.anneeScolaire,
      dateInscription: this.dateInscription,
      statutScolaire: this.statutScolaire,
      classeId: this.classeId,
      eleveDataId: this.eleveDataId,
      niveauClasse: this.niveauClasse,
      classe: this.classe,
      cycle: this.cycle,
      niveauScolaire: this.niveauScolaire,
      nom: this.nom,
      prenom: this.prenom,
      dateNaissance: this.dateNaissance,
      sexe: this.sexe,
      photo: this.photo,
      matricule: this.matricule,
      statut: this.statut,
      lieuDeNaissance: this.lieuDeNaissance,
      contact: this.contact,
      payements: this.payements,
      createdAt: this.createdAt,
      updatedAt: this.updatedAt
    };
  }

  validate() {
    const errors = [];
    if (!this.anneeScolaire) errors.push("L'année scolaire est requise");
    if (!this.dateInscription) errors.push("La date d'inscription est requise");
    if (!this.classeId) errors.push("La classe est requise");
    if (!this.eleveDataId) errors.push("L'élève est requis");
    if (!['nouveau', 'redoublant'].includes(this.statutScolaire)) {
      errors.push('Le statut scolaire doit être nouveau ou redoublant');
    }
    return errors;
  }
}

File: src-tauri/src/models/inscription.rs
use serde::{Deserialize, Serialize};
use chrono::{DateTime, Utc};
use uuid::Uuid;
use sqlx::Type;

#[derive(Debug, Clone, Serialize, Deserialize, Type)]
#[sqlx(type_name = "TEXT", rename_all = "lowercase")]
pub enum StatutScolaire {
    #[serde(rename = "nouveau")]
    Nouveau,
    #[serde(rename = "redoublant")]
    Redoublant,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BaseInscription {
    pub annee_scolaire: String,
    pub date_inscription: String,
    pub statut_scolaire: StatutScolaire,
    pub classe_id: String,
    pub eleve_data_id: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PayementCompact {
    pub id: String,
    pub montant_paye: f64,
    pub statut: String,
    pub date_payement: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Inscription {
    pub id: String,
    pub annee_scolaire: String,
    pub date_inscription: String,
    pub statut_scolaire: StatutScolaire,
    pub classe_id: String,
    pub eleve_data_id: String,
    pub niveau_classe: String,
    pub classe: String,
    pub cycle: String,
    pub niveau_scolaire: String,
    pub nom: String,
    pub prenom: String,
    pub date_naissance: String,
    pub sexe: super::eleve_data::Sexe,
    pub photo: Option<String>,
    pub matricule: String,
    pub statut: super::eleve_data::StatutEleve,
    pub lieu_de_naissance: Option<String>,
    pub contact: Option<String>,
    pub payements: Option<Vec<PayementCompact>>,
    pub created_at: Option<DateTime<Utc>>,
    pub updated_at: Option<DateTime<Utc>>,
}